### PR TITLE
Search refactoring 2

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
#11508

File: sdk/search/azure-search-documents/azure/search/documents/indexes/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+from .._internal import (
+    ComplexField,
+    SearchableField,
+    SimpleField,
+    edm,
```
Comment (bryevdv) [inline sdk/search/azure-search-documents/azure/search/documents/indexes/models/__init__.py:31]: None of these are generated models, should they go here?

Comment (xiangyan99) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/models/__init__.py:31]: We want to put all models under models namespace, not correct?

File: sdk/search/azure-search-documents/tests/test_queries.py [hunk 2]
```diff
@@ -10,13 +10,13 @@
 except ImportError:
     import mock
 
-from azure.search.documents._index._generated.models import (
+from azure.search.documents._internal._generated.models import (
     AutocompleteRequest,
     SearchRequest,
     SuggestRequest,
```
Comment (bryevdv) [inline sdk/search/azure-search-documents/tests/test_queries.py:16]: I think the original import was not ideal, these are available from public API, I think?

Comment (xiangyan99) [reply sdk/search/azure-search-documents/tests/test_queries.py:16]: No they are not. Should we expose them?

File: sdk/search/azure-search-documents/azure/search/documents/indexes/models/__init__.py (hunk 1 above)
Comment (bryevdv) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/models/__init__.py:31]: I might just have a misundertstanding. I understood "models" to be synonymous with "generated models" but if it is' more general than that, then this is fine.

Comment (bryevdv) [reply sdk/search/azure-search-documents/azure/search/documents/indexes/models/__init__.py:31]: I might just have a misunderstanding. I understood "models" to be synonymous with "generated models" but if that is not the case then this is fine

File: sdk/search/azure-search-documents/tests/test_queries.py (hunk 2 above)
Comment (bryevdv) [reply sdk/search/azure-search-documents/tests/test_queries.py:16]: Hrmm actually I was thinking of `AutocompleteQuery` etc, nevermind!

**Pull request merged**